earth_enterprise/src/common/profiler.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 the Open GEE Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <unistd.h>
+#include <sys/syscall.h>
+
+#include "common/profiler.h"
+#include "common/notify.h"
+#include "common/timeutils.h"
+
+using namespace std;
+
+// Initialize static members of Profiler class
+Profiler * const Profiler::_instance = new Profiler();
+
+// Initialize static members of BlockProfiler class
+Profiler * const BlockProfiler::profiler = Profiler::instance();
+
+// Log a profiling message
+void Profiler::log(
+    const string & operation,
+    const string & object,
+    const double startTime,
+    const double endTime,
+    const size_t size) {
+
+  const double duration = endTime - startTime;
+  const pid_t tid = syscall(SYS_gettid);
+  stringstream message;
+
+  message.setf(ios_base::fixed, ios_base::floatfield);
+  message << setprecision(9)
+          << operation << " " << object << ": "
+          << "start time: " << startTime
+          << ", "
+          << "end time: " << endTime
+          << ", "
+          << "duration: " << duration
+          << ", "
+          << "thread: " << tid;
+  if (size > 0) {
+    message << ", size: " << size;
+  }
+
+  notify(NFY_NOTICE, "%s\n", message.str().c_str());
```

Comment:
where log text are stored ? in a file or stdout ? or can this be controlled by passing a 'std::stringstream' as argument ?

Reply:
For now this prints to `stderr`. The idea is that this ticket includes the bare minimum of a logging framework and we can enhance it in future tickets as needed.